src/pytorch_lightning/strategies/ddp.py

```diff
@@ -145,18 +135,12 @@ def _configure_launcher(self) -> None:
         assert self.cluster_environment is not None
         if not self.cluster_environment.creates_processes_externally:
             self._launcher = _SubprocessScriptLauncher(self.cluster_environment, self.num_processes, self.num_nodes)
-            self._rank_0_will_call_children_scripts = True
```

Comment:
When this ran, `_should_run_deadlock_detection` would return True which means that the deadlock detection logic ran for most people using the subprocessscriptlauncher, not just those who set the environment.

However, it's not clear how useful this was. I just wanted to point out that this was enabled more often than we thought (if I'm not missing something)

Reply:
I didn't notice that. I think regardless of this, the deadlock detection does not influence how the program runs in the normal case and so it won't be a breaking change. I will update the description of the PR.

Reply:
Yes. To me, it's a question of "how useful was this for DDP users?"

if it wasn't then we can proceed.
otherwise maybe we should just convert this logic into a plugin that strategies opt-in. Instead of through inheritance where they are forced to override or deal with this logic

Reply:
Created #16410 as a result from our offline chat.